src/server/src/main/java/com/spotify/reaper/resources/view/NodesStatus.java

```diff
@@ -28,34 +29,56 @@
+  private static final List<Pattern> endpointHostIdPatterns = Lists.newArrayList();
+  private static final List<Pattern> endpointTokensPatterns = Lists.newArrayList();
+
+  private static final Pattern endpointNamePattern = Pattern.compile("^([0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3})",
+      Pattern.MULTILINE | Pattern.DOTALL);
+  private static final Pattern endpointStatus22Pattern = Pattern.compile("(STATUS):([0-9]*):(\\w+)");
+  private static final Pattern endpointDc22Pattern = Pattern.compile("(DC):([0-9]*):([0-9a-zA-Z-\\.]+)");
+  private static final Pattern endpointRack22Pattern = Pattern.compile("(RACK):([0-9]*):([0-9a-zA-Z-\\.]+)");
+  private static final Pattern endpointLoad22Pattern = Pattern.compile("(LOAD):([0-9]*):([0-9eE.]+)");
+  private static final Pattern endpointRelease22Pattern = Pattern.compile("(RELEASE_VERSION):([0-9]*):([0-9.]+)");
+  private static final Pattern endpointSeverity22Pattern = Pattern.compile("(SEVERITY):([0-9]*):([0-9.]+)");
+  private static final Pattern endpointHostId22Pattern = Pattern.compile("(HOST_ID):([0-9]*):([0-9a-z-]+)");
+  private static final Pattern endpointTokens22Pattern = Pattern.compile("(TOKENS):([0-9]*)");
+  private static final Pattern endpointStatus21Pattern = Pattern.compile("(STATUS)(:)(\\w+)");
+  private static final Pattern endpointDc21Pattern = Pattern.compile("(DC)(:)([0-9a-zA-Z-\\.]+)");
+  private static final Pattern endpointRack21Pattern = Pattern.compile("(RACK)(:)([0-9a-zA-Z-\\.]+)");
+  private static final Pattern endpointLoad21Pattern = Pattern.compile("(LOAD)(:)([0-9eE.]+)");
+  private static final Pattern endpointRelease21Pattern = Pattern.compile("(RELEASE_VERSION)(:)([0-9.]+)");
+  private static final Pattern endpointSeverity21Pattern = Pattern.compile("(SEVERITY)(:)([0-9.]+)");
+  private static final Pattern endpointHostId21Pattern = Pattern.compile("(HOST_ID)(:)([0-9a-z-]+)");
+
   private static final String NOT_AVAILABLE = "Not available";
-  
+
+  static {
+    initPatterns();
+  }
+
   public NodesStatus(List<GossipInfo> endpointStates) {
     this.endpointStates = endpointStates;
   }
-  
+
   public NodesStatus(String sourceNode, String allEndpointStates, Map<String, String> simpleStates) {
     this.endpointStates = Lists.newArrayList();
-    this.endpointStates.add(parseEndpointStatesString(sourceNode, allEndpointStates, simpleStates)); 
+    this.endpointStates.add(parseEndpointStatesString(sourceNode, allEndpointStates, simpleStates));
   }
 
   private GossipInfo parseEndpointStatesString(String sourceNode, String allEndpointStates, Map<String, String> simpleStates) {
     List<EndpointState> endpoints = Lists.newArrayList();
     Matcher matcher;
-    
+
     String[] strEndpoints = allEndpointStates.split("/");
     Double totalLoad = 0.0;
-    
+
     for(int i=1;i<strEndpoints.length;i++){
       Optional<String> endpoint = Optional.absent();
       Optional<String> hostId = Optional.absent();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "hostId". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)